packages/use-auth-client/src/use-auth-client.ts

```diff
@@ -0,0 +1,172 @@
+   */
+  actorOptions?: ActorConfig;
+
+  idlFactory: IDL.InterfaceFactory;
+
+  canisterId: Principal | string;
+}
+
+/**
+ * Options for the useAuthClient hook
+ */
+export type UseAuthClientOptions = {
+  /**
+   * Options passed during the creation of the auth client
+   */
+  createOptions?: AuthClientCreateOptions;
+  /**
+   * Options passed during the login of the auth client
+   */
+  loginOptions?: AuthClientLoginOptions;
+  /**
+   * Options to create an actor using the auth client identity
+   */
+  actorOptions?: CreateActorOptions;
+};
+
+/**
+ * React hook to set up the Internet Computer auth client
+ * @param {UseAuthClientOptions} options configuration for the hook
+ * @see {@link UseAuthClientOptions}
+ * @param {AuthClientCreateOptions} options.createOptions  - options passed during the creation of the auth client
+ * @param {AuthClientLoginOptions} options.loginOptions -
+ */
+export function useAuthClient(options?: UseAuthClientOptions) {
+  const [authClient, setAuthClient] = React.useState<AuthClient | null>(null);
+  const [identity, setIdentity] = React.useState<Identity | null>(null);
+  const [actor, setActor] = React.useState<Actor | null>(null);
+  const [isAuthenticated, setIsAuthenticated] = React.useState<boolean>(false);
+
+  // load the auth client on mount
+  React.useEffect(() => {
+    AuthClient.create({
+      ...options?.createOptions,
+      idleOptions: {
+        ...options?.createOptions?.idleOptions,
+        onIdle:
+          options?.createOptions?.idleOptions?.onIdle ??
+          (() => {
+            logout();
+          }),
+      },
+    }).then(async client => {
+      setAuthClient(client);
+      setIdentity(client.getIdentity());
+      setIsAuthenticated(await client.isAuthenticated());
+    });
+  }, []);
+
+  React.useEffect(() => {
+    if (identity && options?.actorOptions) {
+      createActor({
+        ...options.actorOptions,
+        agentOptions: { ...options?.actorOptions?.agentOptions, identity },
+      }).then(actor => {
+        setActor(actor);
+      });
+    }
+  }, [identity]);
+
+  /**
+   * Login through your configured identity provider
+   * Wraps the onSuccess and onError callbacks with promises for convenience
+   * @returns {Promise<InternetIdentityAuthResponseSuccess | void>} - Returns a promise that resolves to the response from the identity provider
+   */
+  function login(): Promise<InternetIdentityAuthResponseSuccess | void> {
+    return new Promise((resolve, reject) => {
+      if (authClient) {
+        const callback = options?.loginOptions?.onSuccess;
+        const errorCb = options?.loginOptions?.onError;
+        authClient.login({
+          ...options?.loginOptions,
+          onSuccess: (successResponse?: InternetIdentityAuthResponseSuccess) => {
+            setIsAuthenticated(true);
+            setIdentity(authClient.getIdentity());
+            if (successResponse !== undefined) {
+              callback?.(successResponse);
```

Comment:
same question — why shouldn't the promise resolve here as well?

Reply:
Yes, you're right